umm-spec-lib/src/cmr/umm_spec/migration/service_service_options.clj

```diff
@@ -228,3 +227,122 @@
+   to this new spatial subset structure. Pass back the spatial subset map depending on this type
+   or nil if it is determined that spatial is not supported."
+  [service-type]
+  (case service-type
+        ("OPeNDAP"
+         "THREDDS") {:BoundingBox {:AllowMultipleValues false}}
+        ("ESI"
+         "ECHO ORDERS") {:BoundingBox {:AllowMultipleValues false}
+                         :Shapefile [{:Format "ESRI"}
+                                     {:Format "KML"}
+                                     {:Format "GeoJSON"}]}
+        nil))
+
+(defn- create-variable-subset-type-1_3_4
+  "Create a variable subset map as per UMM-S version 1.3.4 depending on the passed in service type.
+   Looking at the UMM-S records that exist in OPS and UAT, we can migrate existing UMM-S records
+   to this new variable subset structure. Pass back the variable subset map depending on this type."
+  [service-type]
+  (case service-type
+        ("OPeNDAP"
+         "ECHO ORDERS"
+         "ESI"
+         "THREDDS") {:AllowMultipleValues true}
+        {:AllowMultipleValues false}))
+
+(defn- create-subset-type-1_3_4
+  "Create a variable, temporal, or spatial map for the subset element as per UMM-S version 1.3.4
+   depending on the passed in subset type. The passed in service-type will be used to determine what
+   sub elements and values will be used. A map for variable, temporal, or spatial data is passed back."
+  [subset-type service-type]
+  (case subset-type
+        "Variable" {:VariableSubset (create-variable-subset-type-1_3_4 service-type)}
+        "Temporal" {:TemporalSubset {:AllowMultipleValues false}}
+        "Spatial" {:SpatialSubset (create-spatial-subset-type-1_3_4 service-type)}
+        nil))
+
+(defn create-subset-type-1_3_3-to-1_3_4
+  "Create the UMM-S version 1.3.4 subset element using the service type to make choices of which
+   sub elements to use.  A full map that represents the subset structure is passed back."
+  [s]
+  (let [service-type (:Type s)
+        subset-types (distinct
+                       (get-in s [:ServiceOptions :SubsetTypes]))]
+    (->> subset-types
+         (map #(create-subset-type-1_3_4 % service-type))
+         (into {})
+         (remove-nil-keys)
+         (remove-empty-maps))))
+
+(defn find-first-supported-reformatting
+  "Look through the passed in collection which contains the supported reformatings and return the
+   index of the array of the first occurence where the SupportedInputFormat matches the passed in
+   input-format."
+  [input-format coll]
+  (first
+    (keep-indexed #(when (= (:SupportedInputFormat %2) input-format) %1) coll)))
+
+(defn move-supported-formats-to-reformattings-for-1_3_4
+  "Take the passed in input-formats and output-formats and convert the data
+   to the passed in supported-reformattings. Any duplicate data in the
+   output-formats sub element is omitted."
+  [supported-reformattings input-formats output-formats]
+  (let [join-formats (fn [sup-ref input-format]
```

Comment:
This should be broken into its own function

Reply:
So I don't know that I can.  Reduce only allows for 2 functions to be passed in, but I need all three variables that are used in the function.

Reply:
Based on our slack conversation, I will leave this as is.